llvm/lib/MCCAS/MCCASDebugV1.cpp

```diff
@@ -330,6 +351,18 @@ mccasformats::v1::reconstructAbbrevSection(raw_ostream &OS,
         Form = dwarf::Form::DW_FORM_strp;
 
       WrittenSize += encodeULEB128(Form, OS);
+
+      // Dwarf 5: Section 7.4:
+      // The form DW_FORM_implicit_const has to be handled specially. It's
+      // specification contains a third part, which is a signed LEB128 number.
+      // This number is used as the value of the attribute with the
+      // aformentioned form and nothing is stored in the .debug_info section.
+      if (Form == dwarf::Form::DW_FORM_implicit_const) {
+        int64_t ImplicitVal;
+        if (auto Err = Reader.readSLEB128(ImplicitVal))
+          handleAllErrors(std::move(Err));
```

Comment:
the duplication of this block with the one above is a bit concerning; is there a way to factor this into a helper function?